From 7ddce0443b0f7f65fae3a15b7cad7cc71db429eb Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Tue, 25 Nov 2014 16:34:35 -0500
Subject: [fs] vfs: export do_splice_direct() to modules

Message-id: <20141125163435.18115.50192.stgit@warthog.procyon.org.uk>
Patchwork-id: 100923
O-Subject: [PATCH 23/47] vfs: export do_splice_direct() to modules
Bugzilla: 985875
RH-Acked-by: Steve Best <sbest@redhat.com>

From: Miklos Szeredi <mszeredi@suse.cz>

Export do_splice_direct() to modules.  Needed by overlay filesystem.

Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
Original-commit: 1c118596a7682912106c80007102ce0184c77780
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/internal.h b/fs/internal.h
index 656784f..5ed11d9 100644
--- a/fs/internal.h
+++ b/fs/internal.h
@@ -133,12 +133,6 @@ extern int d_set_mounted(struct dentry *dentry);
 extern ssize_t __kernel_write(struct file *, const char *, size_t, loff_t *);
 
 /*
- * splice.c
- */
-extern long do_splice_direct(struct file *in, loff_t *ppos, struct file *out,
-  loff_t *opos, size_t len, unsigned int flags);
-
-/*
  * pipe.c
  */
 extern const struct file_operations pipefifo_fops;
diff --git a/fs/splice.c b/fs/splice.c
index 5938732..e186d80 100644
--- a/fs/splice.c
+++ b/fs/splice.c
@@ -1318,6 +1318,7 @@ long do_splice_direct(struct file *in, loff_t *ppos, struct file *out,
 
  return ret;
 }
+EXPORT_SYMBOL(do_splice_direct);
 
 static int splice_pipe_to_pipe(struct pipe_inode_info *ipipe,
           struct pipe_inode_info *opipe,
diff --git a/include/linux/fs.h b/include/linux/fs.h
index 4cc46be..c43e88b 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -2524,6 +2524,9 @@ extern ssize_t generic_file_splice_write(struct pipe_inode_info *,
   struct file *, loff_t *, size_t, unsigned int);
 extern ssize_t generic_splice_sendpage(struct pipe_inode_info *pipe,
   struct file *out, loff_t *, size_t len, unsigned int flags);
+extern long do_splice_direct(struct file *in, loff_t *ppos, struct file *out,
+  loff_t *opos, size_t len, unsigned int flags);
+
 
 extern void
 file_ra_state_init(struct file_ra_state *ra, struct address_space *mapping);
-- 
1.7.1